### ISSUE TITLE
[Question] Exchange oredict Aluminium/Aluminum

### ISSUE THREAD

**Issue by Gorden211** [WanionCane/UniDict#68]
I don't know where else to ask this but here we go.
Is it possible to exchange oredicts?
When using unidict with ExNihilo Creatio everything works fine. But I want to use the Thermal Foundation Aluminum ingots which are oredicted as ingotAluminum as well as ingotAluminium. The ExNihilo ingots are used in the recreated recipes and I want to change that.

Is there some way to do this? Or generally is there a way to use a specify "all recipes that make itemx should use the specific item modid:itemx as a result"?

Thanks in advance. Feel free to close this ticket whenever you want since it's not an issue.

**Comment by WanionCane** [comment]
probably some tweaking on "S:resourceBlackList" is enough.
but I will take a look =P

**Comment by Gorden211** [comment]
When using the default blacklist with Aluminium and Alumina in it still uses the ingotAluminium of ExNihilo. That's what's confusing to me. Maybe it also has the ingotAluminum oredict, haven't checked yet.

**Comment by WanionCane** [comment]
oh werid, are you using the lastest UniDict version?

**Comment by Gorden211** [comment]
Yes I am using the latest version the twitch launcher suggests.
It probably still uses it because the ExNihilo ingot uses both oredicts, ingotAluminium and ingotAluminum.

**Comment by WanionCane** [comment]
could you try with the previous version?

**Comment by Gorden211** [comment]
I tried with version 2.0 and it still prefers the ExNihilo variation over the Thermal Foundation one.

**Comment by WanionCane** [comment]
ok, try removing the Aluminium | Aluminum from the black list, if it doesn't work, then there is something really wrong =/
also, u sure that this recipe is created by UnIDict?

**Comment by Gorden211** [comment]
![2018-01-04_21 54 22](https://user-images.githubusercontent.com/25781557/34583951-0bbd7ce0-f19a-11e7-9c2e-d8a86cb5b124.png)

I removed Aluminium from the blacklist, now only Alumina, Chrome and Redstone are left.
The recipe still looks like in the screenshot I attached.
Apparently the ingot by Thermal Foundation does not actually use both oredicts and only uses ingotAluminum, so I got that wrong.

**Comment by WanionCane** [comment]
omg.. thanks a lot!
it seems to be broken =/
sorry for making you do all this testing XD

**Comment by Gorden211** [comment]
No problem at all.
I was quite sure I just did something wrong :D I just started out making a pack, since I didn't know what else to do and found this weirdness. I'm not much of a coder/scripter myself so I thought I must have some error somewhere.

Thanks for the help.

**Comment by WanionCane** [comment]
I tried here, this issue isn't happening with the ingot, but it is happening with the dust.
then I added to "S:userOreDictEntries" this:
> dustAluminium-exnihilocreatio:item_ore_aluminium#2

it removed the dust as being dustAluminium, but, other mods managed to create the recipe lol.
probably this aplies for the ingot, unfortunately =/

**Comment by Gorden211** [comment]
Well when I use:

- ingotAluminium-exnihilocreatio:item_ore_aluminium#3

then I get the Thermal Foundation Aluminum ingot I want. Only the dust is problematic as you said.
Using

- dustAluminium-exnihilocreatio:item_ore_aluminium#2

removes all the recipes that make the dust.
Using

- dustAluminum+exnihilocreatio:item_ore_aluminium#2

gives it the correct oredict entry but it still prefers ExNihilo over ThermalFoundation and using both

- dustAluminum+exnihilocreatio:item_ore_aluminium#2
- dustAluminium-exnihilocreatio:item_ore_aluminium#2

kinda works. It still removes most recipes making the dust but I think maybe it works if I adjust the oredict entry of the ExNihilo Ore Chunks to also be Aluminum instead of Aluminium.

Thanks a lot. I will try out and see how it works.

**Comment by Gorden211** [comment]
Yeah I got it to work perfectly!

So overall I used

- ingotAluminium-exnihilocreatio:item_ore_aluminium#3
- dustAluminum+exnihilocreatio:item_ore_aluminium#2
- dustAluminium-exnihilocreatio:item_ore_aluminium#2
- oreAluminum+exnihilocreatio:item_ore_aluminium#1
- oreAluminium-exnihilocreatio:item_ore_aluminium#1
- pieceAluminum+exnihilocreatio:item_ore_aluminium#0
- pieceAluminium-exnihilocreatio:item_ore_aluminium#0

And it works just as before but with the correct dust, ingot and chunks that I want.

I first thought it didn't work out as I wanted because for some reason the config file was reset and the option B:enableSpecificKindSort was set back to false which I use to specify to use the ExNihilo Ores first.
I don't know where I got an error and why it was set back to false but now it works.

Thank you so much, I think I understand how this all works better now.
